pandas/core/dtypes/cast.py

```diff
@@ -196,6 +193,8 @@ def maybe_box_native(value: Scalar) -> Scalar:
         value = int(value)  # type: ignore[arg-type]
     elif is_bool(value):
         value = bool(value)
+    elif isinstance(value, (np.datetime64, np.timedelta64)):
```

Comment:
AFAICT this condition should be equivalent to previous.  There was this comment and followup https://github.com/pandas-dev/pandas/pull/37648/files#r576461677, but the tests seem to contradict by mapping datetime -> datetime (behavior which isn't changed here)

Reply:
dont you need `datetime`, `timedelta` here as well (e.g. python scalar types)? or i suppose these are *already* boxed right so we don't need them?

Reply:
Yep exactly, these are already boxed correctly (and we look to have some test coverage for this, for example in `pandas.tests.frame.methods.test_to_dict::TestDataFrameToDict::test_to_dict_tz`)